src/hooks/form-date-picker/useFormDatePickerProps.tsx

```diff
@@ -81,11 +85,30 @@ export default function useFormDatePickerProps({
           ),
         }}
         fullWidth
+        onBlur={onBlur}
         //we have our own error and helper text state
         error={undefined}
       />
     ),
-    [ariaDescribedby, autocompleteAttributes, className, id, placeholder],
+    [
+      ariaDescribedby,
+      autocompleteAttributes,
+      className,
+      id,
+      onBlur,
+      placeholder,
+    ],
+  )
+
+  const handleChange = React.useCallback(
+    (newDate) => {
+      if (!(newDate instanceof Date) || isNaN(newDate.valueOf())) {
+        onChange(newDate)
+      } else {
+        onChange(newDate)
+      }
+    },
+    [onChange],
   )
```

Comment:
Not much point in `useCallback`ing this as we pass in a new function every render in all three components.